test/json/builder.spec.js

```diff
@@ -63,11 +63,25 @@
             Items: null
           })).to.equal('{}');
         });
-        return it('ignores undefined', function() {
+        it('ignores undefined', function() {
           return expect(build(rules, {
             Items: void 0
           })).to.equal('{}');
         });
+        it('builds document types', function() {
+          var docTypeRules = {
+            type: 'structure',
+            members: {
+              Items: {
+                type: 'structure',
+                document: true
+              }
+            }
+          };
+          expect(build(docTypeRules, {
+            Items: {foo: 'fooValue', bar: null }
+          })).to.equal('{"Items":{"foo":"fooValue","bar":null}}');
```

Comment:
Add other types?
```suggestion
            Items: {strKey: 'str', numKey: 1, boolKey: true, nullKey: null }
          })).to.equal('{"Items":{"strKey":"str","numKey":1,"boolKey":true,"nullKey":null}}');
```

Add complex test cases in a separate step like:
* List: `Items: {strListKey: ['str'], numListKey: [1], boolListKey: [true], nullListKey: [null] }`
* Structure: `Items: {structureKey: {strKey: 'str', numKey: 1, boolKey: true, nullKey: null }}`

Reply:
Ditto for parser tests